src/apis/token-service.test.ts

```diff
@@ -0,0 +1,175 @@
+
+const sampleTokenList = [
+  {
+    address: '0xbbbbca6a901c926f240b89eacb641d8aec7aeafd',
+    symbol: 'LRC',
+    decimals: 18,
+    occurances: 11,
+    aggregators: [
+      'paraswap',
+      'pmm',
+      'airswapLight',
+      'zeroEx',
+      'bancor',
+      'coinGecko',
+      'zapper',
+      'kleros',
+      'zerion',
+      'cmc',
+      'oneInch',
+    ],
+  },
+  {
+    address: '0xc011a73ee8576fb46f5e1c5751ca3b9fe0af2a6f',
+    symbol: 'SNX',
+    decimals: 18,
+    occurances: 11,
+    aggregators: [
+      'paraswap',
+      'pmm',
+      'airswapLight',
+      'zeroEx',
+      'bancor',
+      'coinGecko',
+      'zapper',
+      'kleros',
+      'zerion',
+      'cmc',
+      'oneInch',
+    ],
+  },
+  {
+    address: '0x408e41876cccdc0f92210600ef50372656052a38',
+    symbol: 'REN',
+    decimals: 18,
+    occurances: 11,
+    aggregators: [
+      'paraswap',
+      'pmm',
+      'airswapLight',
+      'zeroEx',
+      'bancor',
+      'coinGecko',
+      'zapper',
+      'kleros',
+      'zerion',
+      'cmc',
+      'oneInch',
+    ],
+  },
+  {
+    address: '0x514910771af9ca656af840dff83e8264ecf986ca',
+    symbol: 'LINK',
+    decimals: 18,
+    occurances: 11,
+    aggregators: [
+      'paraswap',
+      'pmm',
+      'airswapLight',
+      'zeroEx',
+      'bancor',
+      'coinGecko',
+      'zapper',
+      'kleros',
+      'zerion',
+      'cmc',
+      'oneInch',
+    ],
+  },
+  {
+    address: '0x1f573d6fb3f13d689ff844b4ce37794d79a7ff1c',
+    symbol: 'BNT',
+    decimals: 18,
+    occurances: 11,
+    aggregators: [
+      'paraswap',
+      'pmm',
+      'airswapLight',
+      'zeroEx',
+      'bancor',
+      'coinGecko',
+      'zapper',
+      'kleros',
+      'zerion',
+      'cmc',
+      'oneInch',
+    ],
+  },
+];
+
+const sampleToken = {
+  address: '0x514910771af9ca656af840dff83e8264ecf986ca',
+  symbol: 'LINK',
+  decimals: 18,
+  occurances: 11,
+  aggregators: [
+    'paraswap',
+    'pmm',
+    'airswapLight',
+    'zeroEx',
+    'bancor',
+    'coinGecko',
+    'zapper',
+    'kleros',
+    'zerion',
+    'cmc',
+    'oneInch',
+  ],
+};
+
+describe('FetchtokenList', () => {
+  beforeAll(() => {
+    nock.disableNetConnect();
+  });
+
+  afterAll(() => {
+    nock.enableNetConnect();
+  });
+
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  it('should call the tokens api and return the list of tokens', async () => {
+    nock(TOKEN_END_POINT_API)
+      .get(`/tokens/${NetworksChainId.mainnet}`)
+      .reply(200, sampleTokenList)
+      .persist();
+
+    const tokens = await fetchTokenList(NetworksChainId.mainnet);
+
+    expect(tokens).toStrictEqual(sampleTokenList);
+  });
+  it('should call the api to sync tokens and returns nothing', async () => {
+    nock(TOKEN_END_POINT_API)
+      .get(`/sync/${NetworksChainId.mainnet}`)
+      .reply(200)
+      .persist();
+
+    expect(await syncTokens(NetworksChainId.mainnet)).toBeUndefined();
```

Comment:
I would really appreciate inputs on how to add a better testcase for this function.